g2p/api_v2.py

```diff
@@ -0,0 +1,406 @@
+    DEVELOPMENT=1 uvicorn g2p.api_v2:app --reload --port 5000
+- The --reload switch will watch for changes under the directory where it's
+  running and reload the code whenever it changes.
+- DEVELOPMENT=1 tells the API to accept cross-origin requests (i.e. by sending the
+  appropriate CORS headers) from development servers running on localhost, e.g.
+  http://localhost:4200
+
+For deployment, you can use the ORIGIN environment variable to set the
+URL of your application in order to make it accept requests from that
+site.  For instance if you deployed an application that uses it at
+https://my.awesome.site you would set ORIGIN=https://my.awesome.site
+in your environment variables.  This is usually done through an
+environment variable file (or in a dashboard) and will depend on your
+hosting environment.
+
+You can also spin up the API server grade (on Linux, not Windows) with gunicorn:
+    pip install -r requirements.api.txt
+    gunicorn -b 127.0.0.1:5000 -w 4 -k uvicorn.workers.UvicornWorker g2p.api_v2:app
+
+Once spun up, the API will be visible at
+http://localhost:5000/api/v2/docs
+
+"""
+
+import os
+from enum import Enum
+from typing import Dict, List, Tuple, Union
+
+from fastapi import Body, FastAPI, HTTPException, Path, Query
+from fastapi.middleware.cors import CORSMiddleware
+from networkx import shortest_path
+from networkx.algorithms.dag import ancestors, descendants
+from networkx.exception import NetworkXNoPath
+from pydantic import BaseModel, Field
+
+import g2p
+import g2p.mappings.langs as g2p_langs
+from g2p.exceptions import InvalidLanguageCode, NoPath
+from g2p.log import LOGGER
+from g2p.transducer import CompositeTransductionGraph, TransductionGraph
+
+# Create the v2 version of the API
+api = FastAPI(
+    title="Simple G2P API",
+    description="A simple API for the G2P module",
+    version="2.0.0",
+    contact={"email": "dhd@ecolingui.ca"},
+    license_info={
+        "name": "MIT",
+        "url": "https://github.com/roedoejet/g2p/blob/main/LICENSE",
+    },
+)
+
+# Create an "app" that mounts it in the appropriate place
+app = FastAPI()
+app.mount("/api/v2", api)
+middleware_args: Dict[str, Union[str, List[str]]]
+if os.getenv("DEVELOPMENT", False):  # pragma: no cover
+    LOGGER.info(
+        "Running in development mode, will allow requests from http://localhost:*"
+    )
+    # Allow requests from localhost dev servers
+    middleware_args = dict(
+        allow_origin_regex="http://localhost(:.*)?",
+    )
+else:
+    # Allow requests *only* from mt app (or otherwise configured site name)
+    middleware_args = dict(
+        allow_origins=[
+            os.getenv("ORIGIN", "https://readalong-studio.mothertongues.org"),
+        ],
+    )
+app.add_middleware(
+    CORSMiddleware, allow_methods=["GET", "POST", "OPTIONS"], **middleware_args
+)
+
+# All possible language codes
+LanguageNode = Enum("LanguageNode", [(name, name) for name in g2p_langs.LANGS_NETWORK.nodes])  # type: ignore
```

Comment:
```suggestion
LanguageNode = Enum("LanguageNode", sorted((name, name) for name in g2p_langs.LANGS_NETWORK.nodes))  # type: ignore
```
While semantically identical, the `sorted()` means that when you use the "Try it out" button in the API docs, the drop box has languages in order, making them a lot easier to find.
And, the list of available values is also ordered, making it similarly easier to use.